### ISSUE TITLE
Switching between tabs after detach and attach a one of them

### ISSUE THREAD

**Issue by mariaczi** [jimradford/superputty#825]
If we have open a few connection in tabs and one of them has been detached (put in window) and attached again the tab ordering/sorting (or something else) is going out :/ We are not able to switch correctly between tabs.

**To Reproduce**
1. Open some connections in tabs
2. Detach one of the tab to the window
3. Attach the detached again
4. Open new one connection (should be put as a new tab)
5. Try to switch between all tabs in forward and revers.
6. Probably you has been "catch" between atached and detached tab and new opened.

**Expected behavior**
After detaching tab to windows and again attaching it, the tab ordering/sorting should be the same as we see in the "tabs panel".

 - used by me: Windows 7 SP1 with PuTTY

**Comment by Chealer** [comment]
This must be a duplicate of ticket #769 .

**Comment by irstmon** [comment]
I think at least #769 and #765 and #732 are having problems due to the same problem. When switching a session from windowfill to top/left/right/bottom the list stays correct, only 1 session displayed. As soon as you move the session from ANY of this to float there will be a duplicate - not the re-attching is the problem. However this happens just once per session - float/fill again and again will just be 1 set of duplicates. 

As far as I was able to debug this when switching to float ContentRemoved is not called (as far as I can see it IS called when return from Float) - so the list of windows is not correct anymore and contains a set of duplicates. @jimradford maybe you could have a look at this? As I'm far from a programer I'm having problems finding the right place to figure out where the ContentRemoved should be called when going TO Float (as stated above, it seems to be called when coming FROM Float)

**Comment by mariaczi** [comment]
> 
> 
> This is probably a duplicate of ticket #769 . Please clarify how you achieve step 5.

Using defined shortcut keys. Not mouse click on the tab.

**Comment by Chealer** [comment]
> 
> 
> > This is probably a duplicate of ticket #769 . Please clarify how you achieve step 5.
> 
> Using defined shortcut keys. Not mouse click on the tab.

Thank you. This must be a duplicate of ticket #769.
